mitiq/shadows/post_processing.py

```diff
@@ -0,0 +1,141 @@
+from typing import Tuple, List, Union, Any
+
+import numpy as np
+
+PAULI_MAP = {"X": 0, "Y": 1, "Z": 2}
+
+
+def snapshot_state(
+    b_list: List[Union[int, float]], u_list: List[str]
+) -> np.ndarray[Any, Any]:
```

Comment:
A few comments here:

1. Are there better variable names we could use for `b_list` and `u_list`? Always helpful for readability/maintainability if we know what the variable refers to without looking at examples/the actual code.
2. `int` is a subtype of `float`, so `List[float]` should suffice for `b_list`.
3. I think the `np.ndarray[Any, Any]` means a numpy array of any size with any datatype. Is that right? We usually use the `numpy.typing` library for type-hinting numpy related functions, and in this case I think we should use [`numpy.typing.NDArray`](https://numpy.org/devdocs/reference/typing.html#numpy.typing.NDArray) with maybe something like `npt.NDArray[np.complex64]`. See other examples from our codebase [here](https://github.com/search?q=repo%3Aunitaryfund%2Fmitiq%20npt.NDArray&type=code).

Reply:
I changed to NDArray. But NDArray[np.complex_] typing somehow breaks the classical shadow results. Changing to NDArray[Any] solves the problem. I can not figure out what happened, you are very welcome if you want to look into this problem, but please also check the changing type would not destroy the result in the notebook in #1899.

Reply:
Remember to update the `b_list` and `u_list` variable names. Maybe `bits` and `ops` respectively? I'm not totally sure what the best names are, I'll leave that up to you.

Sure enough naming things is [one of the two hardest things in computer science](https://www.martinfowler.com/bliki/TwoHardThings.html).

Reply:
Thanks for the suggestions, Nate!  But actually, those names are aligned with the paper.  Users don't need to look at examples/the actual code.  But they need to know what the theories are actually doing, by either reading paper or learning from any talks, all of them using "b" and "u" actually. I could change them to 

`oneshot_computational_basis_measurement_state_list`
`oneshot_random_unitaries_sampled_from_cl2_list`

but I actually don't thank that is necessary. and naming them by simple names such as bits/states or ops/unitaries actually would be more confusing.  But I'll let you decide whether they are useful, or which one you want to choose!